src/java/org/apache/cassandra/db/commitlog/CommitLog.java

```diff
@@ -684,5 +713,32 @@ public EncryptionContext getEncryptionContext()
         {
             return encryptionContext;
         }
+
+        /**
+         * Returns Direct-IO/non-buffer used for CommitLog IO.
+         * @return Direct-IO used for CommitLog IO
+         */
+        public boolean isDirectIOEnabled()
+        {
+            return diskAccessMode == Config.CommitLogDiskAccessMode.direct_io;
+        }
+
+        /**
+         * Returns MMAP used for CommitLog IO.
+         * @return MMAP used for CommitLog IO
+         */
+        public boolean isMMAPEnabled()
+        {
+            return diskAccessMode == Config.CommitLogDiskAccessMode.mmap;
+        }
+
+        /**
+         * Returns Standard or buffered I/O used for CommitLog IO.
+         * @return Standard or buffered I/O used for CommitLog IO
+         */
+        public boolean isBufferedIOEnabled()
+        {
+            return diskAccessMode == Config.CommitLogDiskAccessMode.standard;
+        }
```

Comment:
Do we need this method here ?

Reply:
It was needed to test Compressed and Encrypted to compare standard vs Direct-IO. It might be required in future but not now.

Reply:
So what about rename to isStandardModeEnable() ?

It seems that there is a little gap between standard and bufferIo, it may need to take some time for you to explain that .
But in my mind , the code should and could explain by itself if you rename to isStandardModeEnable.

Reply:
> So what about rename to isStandardModeEnable() ?
> 
> It seems that there is a little gap between standard and bufferIo, it may need to take some time for you to explain that . But in my mind , the code should and could explain by itself if you rename to isStandardModeEnable.

Agree and self-explanation is good.